osu.Game/Rulesets/UI/FrameStabilityContainer.cs

```diff
@@ -119,93 +115,109 @@ private void updateClock()
             if (parentGameplayClock == null)
                 setClock(); // LoadComplete may not be run yet, but we still want the clock.
 
-            validState = true;
-            requireMoreUpdateLoops = false;
+            // each update start with considering things in valid state.
+            state = PlaybackState.Valid;
 
-            var newProposedTime = parentGameplayClock.CurrentTime;
+            // our goal is to catch up to the time provided by the parent clock.
+            var proposedTime = parentGameplayClock.CurrentTime;
```

Comment:
Actually, now that I look at it again,

```suggestion
            var proposedTime = frameStableClock.IsPaused.Value ? manualClock.CurrentTime : parentGameplayClock.CurrentTime;
```

*sorta kinda* fixes it, but I don't even know where to *begin* justifying that change...

Reply:
As discussed on stream, fixed this in a very similar way. Need confirmation it works for you, as I can't reproduce locally.

Reply:
I can confirm that the latest changes fix the oscillation issue.